packages/replayio/src/utils/browser/launchBrowser.ts

```diff
@@ -42,42 +45,67 @@ export async function launchBrowser(
     throw new Error(`Replay browser not found at: ${browserExecutablePath}`);
   }
 
-  debug(
-    `Launching browser: ${browserExecutablePath} with args:\n`,
-    args.join("\n"),
-    "\n",
-    processOptions
-  );
+  const processes = await findProcess("name", browserExecutablePath);
+  if (processes.length > 0) {
+    const match = processes[0];
 
-  // Wait until the user quits the browser process OR
-  // until the user presses a key to continue (in which case, we will kill the process)
-  const abortControllerForPrompt = new AbortController();
+    debug(`Browser process already running at ${highlight(match.pid)}`);
 
-  const spawnDeferred = spawnProcess(browserExecutablePath, args, processOptions, {
-    onSpawn: () => {
-      if (process.stdin.isTTY) {
-        console.log(`Recording... ${dim("(press any key to stop recording)")}`);
+    console.log(`Recording... ${dim("(quit the Replay Browser to stop recording)")}`);
 
-        prompt({
-          signal: abortControllerForPrompt.signal,
-        }).then(() => {
-          spawnDeferred.data.kill();
-        });
-      } else {
-        console.log(`Recording... ${dim("(quit the Replay Browser to stop recording)")}`);
-      }
-    },
-    printStderr: (text: string) => {
-      debug(stderrPrefix("stderr"), text);
-    },
-    printStdout: (text: string) => {
-      debug(stdoutPrefix("stdout"), text);
-    },
-  });
+    // Ask the browser to open a new tab
+    spawnProcess(browserExecutablePath, args, processOptions);
+
+    // The best we can do in this case is to regularly poll to see when the process exits
```

Comment:
Currently we check the recording log before launching the browser and then compare that against the log that we get after it. I'm worried that this will lead to problems because both CLI jobs will print overlapping recordings - and then u could already pick a recording A to be uploaded in the first CLI and then pick the same in the second CLI. I'd expect that situation to not be handled well today.

I agree that if we want to proceed... this is probably the best we can do. But I also wonder if it's worth proceeding in this case.

Reply:
I talk about this in the Loom. I don't think there's any way around it.

Reply:
What are your thoughts on exiting the CLI with an error when the user tries to do this? It feels like an edge case that the user tries to call the CLI more than once before they close the browser - is it worth introducing more edge cases on our side to accommodate this situation? 🤔

Reply:
@Domiii and @jcmorrow asked for this change, so I feel like you'd probably want to run this by them more than me.

I think we'd need to keep _most of_ this code in place either way, because we'll still need to detect that the runtime is already open.

Reply:
btw if you haven't already watched the Loom attached to this PR, it's worth giving a watch

Reply:
> I think we'd need to keep most of this code in place either way, because we'll still need to detect that the runtime is already open.

Totally 👍 

> btw if you haven't already watched the Loom attached to this PR, it's worth giving a watch

I've already watched it after you first responded to me - I was stalling it to avoid the risk of waking up my wife ;p 

> and then u could already pick a recording A to be uploaded in the first CLI and then pick the same in the second CLI.

What would happen today in this PR if somebody did this?

Reply:
If they tried to upload the same recording twice?

Likely the 2nd attempt would fail and we'd print an error message. TBH I don't think that's a big deal.

Reply:
> If they tried to upload the same recording twice?

Yes.

> Likely the 2nd attempt would fail and we'd print an error message. TBH I don't think that's a big deal.

If that's the case then I'm fine with it - i would just love to make sure that the error would be reported to the user one way or another.

Reply:
Yeah it looks like we just show the duplicate as failed.

I tested in both directions.

![Screenshot 2024-04-18 at 7 08 26 PM](https://github.com/replayio/replay-cli/assets/29597/d20f4a2e-a12a-4f35-bc3a-003ba9e8c37b)

![Screenshot 2024-04-18 at 7 08 47 PM](https://github.com/replayio/replay-cli/assets/29597/29b453bb-ecc9-46e3-ab61-38ac5beaa4d6)